cli/popper/runner_host.py

```diff
@@ -36,225 +38,215 @@ def run(self, step):
 
         log.info(f'[{step["name"]}] {cmd}')
 
-        if self.config.dry_run:
+        if self._config.dry_run:
             return 0
 
         log.debug(f'Environment:\n{pu.prettystr(os.environ)}')
 
-        ecode, _ = pu.exec_cmd(
-            cmd,
-            step_env,
-            self.config.workspace_dir,
-            HostRunner.spawned_processes)
+        pid, ecode, _ = HostRunner._exec_cmd(cmd, step_env,
+                                             self._config.workspace_dir,
+                                             self._spawned_pids)
+        if pid != 0:
+            self._spawned_pids.remove(pid)
+
         return ecode
 
     def stop_running_tasks(self):
-        for p in HostRunner.spawned_processes:
-            log.info(f'Stopping proces {p.pid}')
-            p.kill()
+        for pid in self._spawned_pids:
+            log.info(f'Stopping proces {pid}')
+            os.kill(pid, signal.SIGKILL)
+
+    @staticmethod
+    def _exec_cmd(cmd, env=None, cwd=os.getcwd(), pids=set(), logging=True):
+        pid = 0
+        try:
+            with Popen(cmd, stdout=PIPE, stderr=STDOUT,
+                       universal_newlines=True, preexec_fn=os.setsid,
+                       env=env, cwd=cwd, text=True) as p:
```

Comment:
@ivotron I checked the docs, and it tells that `text=True` option is not supported before Python 3.7. So, we can either keep the previous way of doing `decode('utf-8')` on the output logs or drop py3.6 support. What do you say? I vote for the first one.

Reply:
oh, good catch! We should use `universal_newlines=True` then

Reply:
Yeah. I will change to it. Thanks.